set.mm

```diff
@@ -735822,6 +735867,340 @@ their group (addition) operations are equal for all pairs of elements of
+  Group multiple operation (extension) and cyclic monoids
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d B i x $.  $d F i $.  $d N i x $.  $d X i x $.
+    mulgnngsum.b $e |- B = ( Base ` G ) $.
+    mulgnngsum.t $e |- .x. = ( .g ` G ) $.
+    mulgnngsum.f $e |- F = ( x e. ( 1 ... N ) |-> X ) $.
+    $( Group multiple (exponentiation) operation at a positive integer
+       expressed by a group sum.  (Contributed by AV, 28-Dec-2023.) $)
+    mulgnngsum $p |- ( ( N e. NN /\ X e. B ) -> ( N .x. X ) = ( G gsum F ) ) $=
+      ( vi cn wcel wa cfv c1 cseq co adantr cv cplusg csn cxp cuz elnnuz biimpi
+      cgsu cfz cmpt wceq a1i eqidd simpr fvmptd elfznn fvconst2g syl2an seqfveq
+      weq eqtr4d cvv eqid elfvex eleq2s adantl fmptd gsumval2 mulgnn 3eqtr4rd
+      cbs ) FLMZGBMZNZFEUAOZDPQOFVNLGUBUCZPQZOEDUGRFGCRVMVNKDVOPFVKFPUDOMZVLVKV
+      QFUEUFSZVMKTZPFUHRZMZNZVSDOGVSVOOZWBAVSGGVTDBDAVTGUIUJWBJUKWBAKUSNGULVMWA
+      UMVMVLWAVKVLUMZSUNVMVLVSLMWCGUJWAWDVSFUOLGVSBUPUQUTURVMBVNDEPFVAHVNVBZVLE
+      VAMZVKWFGEVJOBGEVJVCHVDVEVRVMAVTGBDVMVLATVTMWDSJVFVGBVNVPCEFGHWEIVPVBVHVI
+      $.
+
+    $( Group multiple (exponentiation) operation at a nonnegative integer
+       expressed by a groups sum.  This corresponds to the definition in [Lang]
+       p. 6, second formula.  (Contributed by AV, 28-Dec-2023.) $)
+    mulgnn0gsum $p |- ( ( N e. NN0 /\ X e. B )
+                        -> ( N .x. X ) = ( G gsum F ) ) $=
+      ( wcel co cgsu wceq cc0 ex c0 c1 cfz syl6eq cn0 cn wo wi elnn0 mulgnngsum
+      wa c0g cfv oveq1 eqid mulg0 sylan9eq cmpt oveq2 fz10 eqidd mpteq12dv mpt0
+      syl5eq adantr oveq2d gsum0 eqtr4d jaoi sylbi imp ) FUAKZGBKZFGCLZEDMLZNZV
+      HFUBKZFONZUCVIVLUDZFUEVMVOVNVMVIVLABCDEFGHIJUFPVNVIVLVNVIUGZVJEUHUIZVKVNV
+      IVJOGCLVQFOGCUJBCEGVQHVQUKZIULUMVPVKEQMLVQVPDQEMVNDQNVIVNDARFSLZGUNZQJVNV
+      TAQGUNQVNAVSGQGVNVSROSLQFORSUOUPTVNGUQURAGUSTUTVAVBEVQVRVCTVDPVEVFVG $.
+  $}
+
+  ${
+    $d A x $.  $d F i $.  $d X i $.  $d i x $.  $d .x. x $.
+    cycsubm.b $e |- B = ( Base ` G ) $.
+    cycsubm.t $e |- .x. = ( .g ` G ) $.
+    cycsubm.f $e |- F = ( x e. NN0 |-> ( x .x. A ) ) $.
+    cycsubm.c $e |- C = ran F $.
+    $( Characterization of an element of the set of nonnegative integer powers
+       of an element ` A ` of a magma.  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubmel $p |- ( X e. C <-> E. i e. NN0 X = ( i .x. A ) ) $=
+      ( wcel cv wceq cn0 wrex co ovex crn cfv eleq2i wfn wb fvelrnb ax-mp oveq1
+      fnmpti fvmpt eqeq1d eqcom syl6bb rexbiia 3bitri ) IDNIGUAZNZFOZGUBZIPZFQR
+      ZIURBESZPZFQRDUPIMUCGQUDUQVAUEAQAOZBESZGVDBETLUIFQIGUFUGUTVCFQURQNZUTVBIP
+      VCVFUSVBIAURVEVBQGVDURBEUHLURBETUJUKVBIULUMUNUO $.
+
+    $d A i $.  $d B i $.  $d .x. i $.
+    $( The set of nonnegative integer powers of an element ` A ` of a magma
+       contains ` A ` .  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubmcl $p |- ( A e. B -> A e. C ) $=
```

Comment:
Sorry I didn't cacth this earlier, but in both theorems, you should write "unital magma". Indeed, the formal statement is still true without assuming it, because df-gsum has to give a value for the empty sum even when the structure is not unital, but this does not really make sense. You may also write some explanation like that in your theorems, to at least warn the reader.

Reply:
OK, I will revise the comment accordingly.